projects/noloan/app/src/main/java/hasadna/noloan/DBMessagesHolder.java

```diff
@@ -0,0 +1,100 @@
+package hasadna.noloan;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+// Simple singleton to hold the suggestions list
+public class DBMessagesHolder {
+  private static DBMessagesHolder instance;
+  private List<SmsMessage> spam;
+  private List<SmsMessage> suggestions;
+
+  MessagesListener spamListener;
+  MessagesListener suggestionsListener;
+
+  public static DBMessagesHolder getInstance() {
+    if (instance == null) {
+      instance = new DBMessagesHolder();
+    }
+    return instance;
+  }
+
+  public DBMessagesHolder() {
+    spam = new ArrayList<>();
+    suggestions = new ArrayList<>();
+  }
+
+  public void init(List<SmsMessage> spam, List<SmsMessage> suggestions) {
+    this.spam = spam;
+    this.suggestions = suggestions;
+  }
+
+  public List<SmsMessage> getSpam() {
+    return spam;
+  }
+
+  public List<SmsMessage> getSuggestions() {
+    return suggestions;
+  }
+
+  public void addSpam(SmsMessage spam) {
+    this.spam.add(spam);
+
+    if (spamListener != null) {
+      spamListener.messageAdded();
+    }
+  }
+
+  public void spamModified(SmsMessage spam) {
+    int index = this.spam.indexOf(spam);
```

Comment:
How does `indexOf` find the correct index?
If `spam` was modified, its value is no longer as it was. It looks like `List` uses `hashCode()` (as is usual), so the hashcode of `spam` is changed when it gets modified.

Reply:
This was part of the code @gulilo  added before.

As for my understanding, this function isn't relevant as we don't have any scenario of editing a specific message, just adding / removing.

No?